zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxnSocketNIO.java

```diff
@@ -289,6 +289,10 @@ void connect(InetSocketAddress addr) throws IOException {
             LOG.error("Unable to open socket to " + addr);
             sock.close();
             throw e;
+        } catch (RuntimeException e) {
+            LOG.error("Unable to open socket to " + addr);
```

Comment:
This code looks like the code above.
Can we use multicatch syntax?

Reply:
No, because of Java 6 in this version. Sorry, but I have to use different catch blocks...

Reply:
Zookeeper is java8+ 
As you can see we use lambdas and other language features

Reply:
Not this version. This is 3.4.14.
```
    <!-- maven properties -->
    <maven.compiler.source>1.6</maven.compiler.source>
    <maven.compiler.target>1.6</maven.compiler.target>
```
https://github.com/apache/zookeeper/blob/branch-3.4.14/pom.xml#L265

Reply:
We still use Zookeeper 3.4.13 because our Zookeeper Server is the same version. We can switch on 3.4.14, but I am not sure about more recent versions like 3.5 or 3.6. I fear that client library of the version 3.5 or 3.6 will not work with Zookeeper Server 3.4...